skills_ml/ontologies/esco.py

```diff
@@ -0,0 +1,78 @@
+from .base import Competency, Occupation, CompetencyOntology
+import logging
+import csv
+import requests
+
+lang = '&language=en'
+api_tax = 'https://ec.europa.eu/esco/api/resource/taxonomy?uri=http://data.europa.eu/esco/concept-scheme/isco&language=en'
+
+
+class Esco(CompetencyOntology):
+    def __init__(self):
+        super().__init__()
+        self.is_built = False
+        self.competency_framework.name = 'esco_skills'
+        self.competency_framework.description = 'ESCO Knowledge, Skills/Competencies'
+        self._build()
+    
+    def _build(self):
+        if not self.is_built:
+            reqTax = requests.get(api_tax)
+            getConcepts = reqTax.json()['_links']['hasTopConcept']
+            for concept in getConcepts:
```

Comment:
It would be nice to also log each concept name or id before we process it. Since this is the outer loop, it would be even better to use `enumerate(getConcepts)` to get the concept index and log that and the length so we know how far we are in the processing. Example:

```
num_concepts = len(getConcepts):
for i, concept in enumerate(getConcepts):
    logging.info('Processing concept %s of %s: %s', i, num_concepts, concept['href'])
    ...
```